wallet/src/main/java/com/android/identity_credential/wallet/navigation/WalletDestination.kt

```diff
@@ -0,0 +1,167 @@
+package com.android.identity_credential.wallet.navigation
+
+import androidx.navigation.NamedNavArgument
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavType
+import androidx.navigation.navArgument
+
+
+sealed class WalletDestination(val routeEnum: Route) : DestinationArguments() {
+
+    // String representation of this Destination's route
+    val route = routeEnum.routeName
+    val routeWithArgs = routeEnum.routePathWithArguments
+
+    // Screens with no arguments
+    object Main : WalletDestination(Route.MAIN)
+    object About : WalletDestination(Route.ABOUT)
+    object AddToWallet : WalletDestination(Route.ADD_TO_WALLET)
+    object ProvisionCredential : WalletDestination(Route.PROVISION_CREDENTIAL)
+
+
+    // Screens with arguments
+    object CredentialInfo : WalletDestination(Route.CREDENTIAL_INFO) {
```

Comment:
I don't have much experience with this, but why would not we just do this case (CredentialInfo) with a class instead of a singleton? And then that class could hold the arguments as regular read-only fields?

Reply:
I was mostly interested in rationale. In cases like that it is nice to put it as a comment somewhere (even if it is lengthy).

Reply:
somewhere in the code, I mean